src/endpoint.h

```diff
@@ -242,30 +238,29 @@ class UdpEndpoint : public Endpoint {
     int open_ipv6(const char *ip, unsigned long port, UdpEndpointConfig::Mode mode);
 
     ssize_t _read_msg(uint8_t *buf, size_t len) override;
+
+private:
+    bool is_ipv6;
+    struct sockaddr_in sockaddr;
+    struct sockaddr_in6 sockaddr6;
 };
 
 class TcpEndpoint : public Endpoint {
 public:
     TcpEndpoint(std::string name);
     ~TcpEndpoint() override;
 
+    int write_msg(const struct buffer *pbuf) override;
+    int flush_pending_msgs() override { return -ENOSYS; }
+    bool is_valid() override { return _valid; };
+    bool is_critical() override { return false; };
+
     int accept(int listener_fd);        ///< accept incoming connection
     bool setup(TcpEndpointConfig conf); ///< open connection and apply config
     bool reopen();                      ///< re-try connecting to the server
     void close();
 
-    int write_msg(const struct buffer *pbuf) override;
-    int flush_pending_msgs() override { return -ENOSYS; }
-
-    struct sockaddr_in sockaddr;
-    struct sockaddr_in6 sockaddr6;
-    bool ipv6;
-    int retry_timeout = 0; // disable retry by default
-
-    inline std::string get_ip() { return _ip; }
-    inline unsigned long get_port() const { return _port; }
-    bool is_valid() override { return _valid; };
-    bool is_critical() override { return false; };
+    void set_retry_timeout(int retry_timeout) { _retry_timeout = retry_timeout; };
```

Comment:
This method doesn't seem to be needed as it can only be set by the config

Reply:
See above.